junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/extension/TempDirectoryCleanupTests.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2015-2021 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.engine.extension;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.junit.jupiter.api.io.CleanupMode.ALWAYS;
+import static org.junit.jupiter.api.io.CleanupMode.NEVER;
+import static org.junit.jupiter.api.io.CleanupMode.ON_SUCCESS;
+import static org.junit.platform.engine.discovery.DiscoverySelectors.selectMethod;
+import static org.junit.platform.launcher.core.LauncherDiscoveryRequestBuilder.request;
+
+import java.io.File;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.CleanupMode;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.engine.AbstractJupiterTestEngineTests;
+import org.junit.platform.launcher.LauncherDiscoveryRequest;
+
+/**
+ * Test that {@link TempDir temporary directories} are not deleted if set for {@link CleanupMode#NEVER},
+ * deletes any if set for {@link CleanupMode#ON_SUCCESS} only if the test passes,
+ * and deletes any if set for {@link CleanupMode#ALWAYS}.
+ *
+ * @since 5.9
+ *
+ * @see CleanupMode
+ * @see TempDir
+ */
+class TempDirectoryCleanupTests extends AbstractJupiterTestEngineTests {
+
+	private static File defaultDir;
+	private static File neverDir;
+	private static File alwaysDir;
+	private static File onSuccessFailingDir;
+	private static File onSuccessPassingDir;
```

Comment:
Please add an `@AfterAll` method and delete these directories there if they exist.

Reply:
Done